### ISSUE TITLE
Clarify network/firewall/port requirements in Documentation

### ISSUE THREAD

**Issue by eponerine** [Azure/aks-hci#97]
Assume you have the following scenario:

- You have 2 isolated VLANs - `VLAN 100` for **Fabric Management** and `VLAN 200` for **k8s-related stuff**. They cannot route to each other by default due to security concerns.

- You configure the **Physical ASHCI Nodes** on the Fabric Management Network `VLAN 100`, including the Failover Cluster's access IP and the VM hosting Windows Admin Center.

- You deploy your AKS-on-HCI VMs on `VLAN 200`, meaning the IP pool for the control plane VMs, worker node VMs, and the VIPs for deployed services.

Based on this configuration, you'll need to configure your Cloud Agent Service to be an IP on `VLAN 100`, which is easy enough to configure, but as mentioned earlier, they cannot route to each other by default.

_**What firewall rules and allowed ports would we need to configure to get deployment to work?**_ 

I assume that WAC needs to access the k8s API endpoint... but my Security Team isn't gonna let us do ANY-ANY between these VLANs.

Once answered here, can we update the Documentation to include this info?

**Comment by comnam90** [comment]
Agree, communication requirements need enhancing to allow for review with security teams.
I'd still love to see use of the Hyper-V VMBus for communication between components where possible to reduce network connectivity requirements

**Comment by Elektronenvolt** [comment]
I'm using:

**TCP:**
Ports 46000 and 6443 are ports on the VMs that need to be reached by the physical host.
And port 22 to collect logs with _Get-AksHciLogs_
-> source: `VLAN 100`  destination: `VLAN 200` 

Ports 65000, 55000 are on the physical host and need to be reachable by processes in the AKS VMs.
-> source: `VLAN 200`  destination: `VLAN 100` 

I don't have seen drops at the firewall the last month, let me know in case I missed a port.

**Comment by eponerine** [comment]
@comnam90 - yes! Couldn't agree more with you about leveraging VMBus for this type of operations.

@Elektronenvolt - thank you for the clarity; I have my team working thru that now to confirm it works and will reply back. Hopefully the Docs team can update soon after :)

**Comment by eponerine** [comment]
@elektronvolt - are you sure those ports are still accurate?

![image](https://user-images.githubusercontent.com/503974/119197857-689cea80-ba56-11eb-92b7-553e47bf9605.png)

**Comment by Elektronenvolt** [comment]
@eponerine well, I don't have it on an Azure Stack HCI right now. It's on a single node, the last WAC deployment I've tried over a month ago and since then I'm using powershell (again).  I got these ports from Zach Wachtel in Feb, that I need 22 I've seen after running  _Get-AksHciLogs_ the first time.
I see now that 45001 is not open at my firewall, but in the docs: [network-port-and-url-requirements](https://docs.microsoft.com/en-us/azure-stack/aks-hci/system-requirements#network-port-and-url-requirements), I'll watch out for drops with the next setup.

If I start now the [April release](https://github.com/Azure/aks-hci/releases/tag/AKS-HCI-2104) on a single node over powershell, TCPview says port 6443 is required from `vlan100` to `vlan200`
![image](https://user-images.githubusercontent.com/6702311/119201812-e4ffef80-ba8f-11eb-8ea8-95a491c411e0.png)

**Comment by Elektronenvolt** [comment]
A two node Windows Server failover cluster setup with the May 2021 1.0 release worked with my firewall settings.

**Comment by eponerine** [comment]
@mkostersitz - it appears that this page now makes no reference to ports 22, 6443, or 46000
https://docs.microsoft.com/en-us/azure-stack/aks-hci/system-requirements?tabs=allow-table#network-port-and-url-requirements

Have those ports been deprecated? I doubt it because 6443 is still what the CA is listening on with LB VIP. Can we get some clarity on Docs please?

**Comment by mkostersitz** [comment]
Thanks @eponerine for raising awareness. @abhilashaagarwala please review and make corrections as needed.

**Comment by Elektronenvolt** [comment]
@abhilashaagarwala @eponerine - I think the concept of having the physical nodes in a separate vlan than Aks-Hci (and a firewall between) is not in the docs. That's why 22, 6443, 46000 don't show up there.

We still use these [FW rules](https://github.com/Azure/aks-hci/issues/97#issuecomment-845727932) since May 2021.

Ports 45000, 45001 are not open at our FW, looks like there are connections within the same vlan only.

**Comment by nmdange2** [comment]
Something else to note on firewall rules, if the management cluster is on a different VLAN from a workload cluster, the management cluster also appears to need access to the workload cluster.

**Comment by abhilashaagarwala** [comment]
Hey everyone! Thank you all for contributing!! 

@Elektronenvolt - If I understand correctly, your recommendation is to include 22, 6443 and 46000 as additional ports in our documentation here - https://docs.microsoft.com/en-us/azure-stack/aks-hci/system-requirements?tabs=allow-table#network-port-and-url-requirements

We then need to revise WAC UI to make sure the right ports are listed in the first UI page.

**Comment by Elektronenvolt** [comment]
Hi @abhilashaagarwala

This section of the docs show opened ports at the Windows Defender  Firewall at the Hyper-V host machines.
![image](https://user-images.githubusercontent.com/6702311/166198860-fcd13161-70eb-4ff0-80f8-1c602ba06363.png)

The scenario @eponerine describes and we use as well is about FW ports between the Hyper-V host machines vLan and the AKS-HCI VMs (+cloudagent) vlan. I think this scenario + FW ports should be added to the docs, looks like this is common setup.

**Comment by abhilashaagarwala** [comment]
Hey everyone, here is the link to the updated documentation  - https://docs.microsoft.com/en-us/azure-stack/aks-hci/system-requirements?tabs=allow-table#aks-on-azure-stack-hci-requirements

Huge shout out to @Elektronenvolt for helping with this!!

**Comment by abhilashaagarwala** [comment]
Please lmk in 1-2 days if something else is needed here.

**Comment by eponerine** [comment]
@abhilashaagarwala - this is a great start, but may be difficult to parse for non-English speaking folks. I think it makes sense to make the table look something like this. Please update port 46000 with whatever the actual service name is:

![image](https://user-images.githubusercontent.com/503974/166569313-231d4cc4-d287-416c-8584-f6aed4d201eb.png)


| Port  | Source                 | Description                                          | Firewall Notes                                                                               |
|-------|------------------------|------------------------------------------------------|----------------------------------------------------------------------------------------------|
| 22    | AKS VMs                | Required to collect logs when using Get-AksHciLogs.  | If using separate VLANs, the physical Hyper-V Hosts need to access the AKS VMs on this port. |
| 6443  | AKS VMs                | Required to communicate with Kubernetes APIs.        | If using separate VLANs, the physical Hyper-V Hosts need to access the AKS VMs on this port. |
| 45000 | Physical Hyper-V Hosts | wssdAgent gRPC Server                                | No cross-VLAN rules are needed.                                                              |
| 45001 | Physical Hyper-V Hosts | wssdAgent gRPC Authentication                        | No cross-VLAN rules are needed.                                                              |
| 46000 | AKS VMs                | ???                                                  | If using separate VLANs, the physical Hyper-V Hosts need to access the AKS VMs on this port. |
| 55000 | Cluster Resource       | wssdCloudAgent gRPC Server                           | If using separate VLANs, the AKS VMs need to access the Cluster Resource's IP on this port.  |
| 65000 | Cluster Resource       | wssdCloudAgent gRPC Authentication                   | If using separate VLANs, the AKS VMs need to access the Cluster Resource's IP on this port.  |

**Comment by Elektronenvolt** [comment]
Yes, great table - the Cloud Resource's IP with Port 55000 and 65000 - it's the `-cloudservicecidr` from command `Set-AksHciConfig`. Better to call it Cloud Agent IP instead?

**Comment by eponerine** [comment]
I agree, but was trying to be as literal as possible for folks to understand where the thing lives. "Cloud Agent IP" is somewhat ambiguous. I updated table to include a mix of info.

**Comment by eponerine** [comment]
@Elektronenvolt and @abhilashaagarwala .... what is 46000 actually used for? I still can't figure it out!

**Comment by Elektronenvolt** [comment]
@eponerine - checked my E-mail conversations and network traces. Don't know what it's used for, but I have a confirmation (June 2021) that its a port at the VMs that needs to be reached by the physical host.

**Comment by abhilashaagarwala** [comment]
<img width="547" alt="image" src="https://user-images.githubusercontent.com/60489035/170405593-8592a0f0-ba4c-4e2c-95e7-c528f659eb65.png">
Does this answer your question?

**Comment by Elektronenvolt** [comment]
@abhilashaagarwala  - yes it does, thanks!
